### PR TITLE
Allocator interface

### PR DESCRIPTION
**Pull request by lpabon**
Signed-off-by: Luis Pabón <luis@portworx.com>

### CODE REVIEW

File: pkg/allocator/roundrobin/roundrobin.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	return &Allocator{}
+}
+
+// DetermineNodeToAddStorage returns a node on the cluster which will be used to add storage
+// TODO: This will be an inteface to a new algorithm object
+func (r *Allocator) DetermineNodeToAddStorage(
+	t *topology.Topology,
+	class *config.Class,
+) *topology.StorageNode {
+	node := t.Cluster.StorageNodes[0]
```
Comment (sdeoras) [inline pkg/allocator/roundrobin/roundrobin.go]: not sure if `t` is validated elsewhere but i would check for nil pointers before dereferencing and accessing indices

File: pkg/allocator/roundrobin/roundrobin.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	class *config.Class,
+) (*topology.StorageNode, *topology.Pool, *topology.Device) {
+	var (
+		node   *topology.StorageNode
+		device *topology.Device
+		pool   *topology.Pool
+	)
+
+	// Get the node
+	for _, currentNode := range t.Cluster.StorageNodes {
```
Comment (sdeoras) [inline pkg/allocator/roundrobin/roundrobin.go]: same comment here. is `t` and tree underneath it validated to be not nil?

File: pkg/allocator/roundrobin/roundrobin.go (hunk 1 above)
Comment (lpabon) [reply pkg/allocator/roundrobin/roundrobin.go]: good catch.

File: pkg/allocator/roundrobin/roundrobin.go (hunk 2 above)
Comment (lpabon) [reply pkg/allocator/roundrobin/roundrobin.go]: Yeah, those are good.

**Pull request merged**